taskchampion/src/server/cloud/server.rs

```diff
@@ -948,6 +948,29 @@ mod tests {
         );
     }
 
+    #[test]
+    fn get_child_version_single_invalid() {
+        // This is a regression test for #387.
+        let mut server = make_server();
+        let (v1, v2, v3) = (Uuid::new_v4(), Uuid::new_v4(), Uuid::new_v4());
+        // Here v2 is latest, so v1 is not a valid child of v2.
+        server.mock_add_version(v2, v1, 1000, b"first");
+        server.mock_add_version(v3, v2, 1000, b"second");
```

Comment:
```suggestion
        server.mock_add_version(v1, v2, 1000, b"second");
        server.mock_add_version(v2, v3, 1000, b"third");
```
If I understand this correctly, this ordering and naming makes more sense:
- The fourth `data` argument is associated with the second `child` argument, right?
- Children are newer than their parents, right? So (as the first assertion demonstrates) the first version is `v3` and the last version is `v1`. That seems confusing.

Reply:
Huh, I thought all of the tests had this backward ordering, but now that I look again that's not the case at all.

Reply:
That sounds familiar. I think maybe it was like that until i complained. :)